tlsobs-api/handlers.go

```diff
@@ -430,6 +431,70 @@ func PathsHandler(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+// TruststoreHandler handles the /truststore endpoint of the api.
+// It queries the database for all certificates trusted by a certain program.
+// It takes the following parameters as HTTP query parameters:
+//     store: one of {"mozilla", "android", "apple", "microsoft", "ubuntu"}
+//     format: one of {"json", "pem"}
+func TruststoreHandler(w http.ResponseWriter, r *http.Request) {
+	setResponseHeader(w)
+	log := logger.GetLogger()
+	log.WithFields(logrus.Fields{
+		"form values": r.Form.Encode(),
+		"headers":     r.Header,
+		"uri":         r.URL.RequestURI(),
+	}).Debug("Truststore Endpoint received request")
+
+	val := r.Context().Value(dbKey)
+	if val == nil {
+		httpError(w, http.StatusInternalServerError, "Could not find database handler in request context")
+		return
+	}
+	db := val.(*pg.DB)
+	certs, err := db.GetAllCertsInStore(r.FormValue("store"))
+	if err == pg.ErrInvalidCertStore {
+		httpError(w, http.StatusBadRequest, "Invalid certificate trust store provided")
+		return
+	} else if err != nil {
+		logrus.Error("Error querying truststore:", err)
+		httpError(w, http.StatusBadRequest, "Invalid certificate trust store provided")
+		return
+	}
+	switch r.FormValue("format") {
+	case "json":
+		certsJSON, err := json.Marshal(certs)
+		if err != nil {
+			httpError(w, http.StatusInternalServerError, "Could not marshal certificates")
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/x-pem-file")
+		w.Write(certsJSON)
+		logRequest(r, http.StatusOK, len(certsJSON))
+	case "pem":
+		var buffer bytes.Buffer
+		for _, cert := range certs {
+			x509, err := cert.ToX509()
+			if err != nil {
+				httpError(w, http.StatusInternalServerError, "Could not convert certificate to X509")
+				return
+			}
+			err = pem.Encode(&buffer, &pem.Block{Type: "CERTIFICATE", Bytes: x509.Raw})
+			if err != nil {
+				httpError(w, http.StatusInternalServerError, "Error PEM-encoding certificate")
+				return
+			}
+			buffer.Write([]byte{'\n'})
+		}
+		w.Header().Set("Content-Type", "application/x-pem-file")
```

Comment:
let's use `text/plain`, I don't want browsers to try to import those api responses automatically.